drivers/gpio/gpio_xlnx_axi.c

```diff
@@ -32,15 +32,14 @@
 #define MAX_GPIOS 32
 
 struct gpio_xlnx_axi_config {
-	/* gpio_driver_config needs to be first */
+	DEVICE_MMIO_ROM;
 	struct gpio_driver_config common;
```

Comment:
`DEVICE_MMIO_ROM` and `DEVICE_MMIO_RAM` also have a requirement that they need to be the first member of the struct, so perhaps it would have been better to not rely on the use of those macros and to add address, size and mmio address information to the struct (omitting mmio address and size if `CONFIG_MMU` is not defined.